app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+            // TODO All of these data sources are unknown, I think they may be from partner information, but don't know for sure
+            partnerOptInSms = unknown.partnerOptInSms,
+            partnerOptInEmail = unknown.partnerOptInEmail,
+            partnerOptInVolunteer = unknown.partnerOptInVolunteer,
+            finishWithState = true, // TODO is this a default value?
+            createdViaApi = true, // TODO is this a default value?
+            sourceTrackingId = unknown.sourceTrackingId,
+            partnerTrackingId = unknown.partnerTrackingId,
+            geoLocation = unknown.geoLocation,
+            openTrackingId = unknown.openTrackingId,
+            voterRecordsRequest = buildVoterRecordsRequest()
+        )
+
+        return RockyRequest(body)
+    }
+
+    private fun buildVoterRecordsRequest(): VoterRecordsRequest {
+        return VoterRecordsRequest(
+            type = "registration", // TODO is this a default value?
+            // TODO is this from somewhere else, I can't find a source?
+            // TODO Does this _have_ to be a full datetime stamp? It is in the example, but I don't see a formatter anywhere in the project
+            //  Looks like not? From old RockyRequest.java: Date generatedDate = Dates.parseISO8601_Date(Db.getString(cursor, GENERATED_DATE));
+            generatedDate = Dates.formatAsISO8601_Date(Date()),
+            canvasserName = partnerInformation.canvasserName,
+            voterRegistration = buildVoterRegistration()
+        )
+    }
+
+    private fun buildVoterRegistration(): VoterRegistration {
+        val mailingAddress = if (addressData.isMailingAddressDifferent) {
+            addressData.mailingAddress ?: throw InvalidRegistrationException(
+                "addressData.isMailingAddressDifferent is null",
+                R.string.different_mailing_address_not_filled_out
+            )
+        } else {
+            null
+        }
+
+        // TODO Is this the way that we should be specifying race? It's required in the API, but not in the app
+        val race = (additionalInfoData.race ?: Race.DECLINE).toString()
+
+        val party = when (additionalInfoData.party) {
+            Party.OTHER_PARTY -> additionalInfoData.otherPoliticalParty
+                ?: throw InvalidRegistrationException(
+                    "otherPoliticalParty is null",
+                    R.string.must_write_in_party
+                )
+            else -> additionalInfoData.party.toString()
+        }.toLowerCase() // TODO i
+
+        // TODO NO_WRAP was specified in ApiSignature, but probably should be in the API docs as well
+        val signature = Base64.encodeToString(reviewData.signature, Base64.NO_WRAP)
+
+        return VoterRegistration(
+            registrationHelper = buildRegistrationHelper(),
+            dateOfBirth = Dates.formatAsISO8601_ShortDate(newRegistrationData.birthday), //TODO DateAdapter used short date, is this correct?
```

Comment:
yes, i know it's not consistent, but that's just how it worked out. We'll double check all of this on the QA server when we're done